src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -124,6 +136,25 @@ public static IResourceBuilder<AzureCosmosDBEmulatorResource> WithGatewayPort(th
         });
     }
 
+    /// <summary>
+    /// Configures the partition count for the Azure Cosmos DB emulator.
+    /// </summary>
+    /// <param name="builder">Builder for the Cosmos emulator container</param>
+    /// <param name="count">Desired partition count.</param>
+    /// <returns>Cosmos emulator resource builder.</returns>
+    /// <remarks>Not calling this method will result in the default of 25 partitions. The actual started partitions is always one more than specified.
+    /// See <a href="https://learn.microsoft.com/en-us/azure/cosmos-db/emulator-windows-arguments#change-the-number-of-default-containers">this documentation</a> about setting the partition count.
+    /// </remarks>
+    public static IResourceBuilder<AzureCosmosDBEmulatorResource> WithPartitionCount(this IResourceBuilder<AzureCosmosDBEmulatorResource> builder, int count)
+    {
+        if (count < 1 || count > 250)
+        {
+            throw new ArgumentOutOfRangeException(nameof(count), "Count must be between 1 and 250.");
```

Comment:
You could use the overload that provides the value that's out of range:

```suggestion
            throw new ArgumentOutOfRangeException(nameof(count), count, "Count must be between 1 and 250.");
```